### PR TITLE
Pr 72292- Add context to inventory plugins

### PR DESCRIPTION
**Pull request by samccann**
##### SUMMARY

I tried to rebase PR 72292 and made an unholy mess of things. So going to close that PR in favor of this one.

Rebase of #72292
##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

docs/docsite/rst/plugins/inventory.rst
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Review by ariordan-redhat — APPROVED** [review]
LGTM

File: docs/docsite/rst/plugins/inventory.rst [hunk 1]
```diff
@@ -80,12 +87,24 @@ You can create dynamic groups using host variables with the constructed ``keyed_
...
     groups:
       # add hosts to the group development if any of the dictionary's keys or values is the word 'devel'
       development: "'devel' in (tags|list)"
+      # add hosts to the "private_only" group if the host doesn't have a public IP associated to it
+      private_only: "public_ip_address is not defined"
     compose:
-      # set the ansible_host variable to connect with the private IP address without changing the hostname
-      ansible_host: private_ip_address
+      # use a private address where a public one isn't assigned
+      ansible_host: public_ip_address is defined | ternary(public_ip_address, private_ip_address)
```
Comment (bcoca) [inline docs/docsite/rst/plugins/inventory.rst:101]: this wont work as the arguments to ternary must be defined

```ansible_host: public_pi_address|default(private_ip_address)```

Comment (samccann) [reply docs/docsite/rst/plugins/inventory.rst:101]: doh! Sorry about that. PR to fix - #78186

**Pull request merged**